File: sharedModule/src/main/java/ru/nekit/android/qls/pupil/avatar/PupilAvatarConverter.java
package ru.nekit.android.qls.pupil.avatar;


import android.support.annotation.NonNull;
import android.text.TextUtils;

import java.util.ArrayList;
import java.util.List;

import ru.nekit.android.qls.pupil.Pupil;
import ru.nekit.android.qls.pupil.PupilSex;

public class PupilAvatarConverter {

    private static final String PART_AND_VARIANT_DELIMITER = ":";
    private static final String DELIMITER = ";";

    public static String toString(@NonNull IPupilAvatarPart[] parts,
                                  @NonNull int[] partVariantPositions) {
        List<String> stringList = new ArrayList<>();
        int index = 0;
        for (IPupilAvatarPart pupilAvatarPart : parts) {
            stringList.add(String.format("%s%s%s", pupilAvatarPart.getPartName(),
                    PART_AND_VARIANT_DELIMITER, partVariantPositions[index]));
            index++;
        }
        return TextUtils.join(DELIMITER, stringList);
    }

    public static List<PupilAvatarPartAndVariant> toPupilAvatarPartAndVariant(
            @NonNull Pupil pupil) {
        List<PupilAvatarPartAndVariant> result = new ArrayList<>();
        if (pupil.avatar != null) {
            String[] pupilAvatarPartStrings = TextUtils.split(pupil.avatar, DELIMITER);
            for (String pupilAvatarPartString : pupilAvatarPartStrings) {
                String[] pupilAvatarPartStringValues =
                        TextUtils.split(pupilAvatarPartString, PART_AND_VARIANT_DELIMITER);
                IPupilAvatarPart[] pupilAvatarParts = pupil.sex == PupilSex.BOY ? PupilBoyAvatarPart.values()
                        : PupilGirlAvatarPart.values();
                for (IPupilAvatarPart pupilAvatarPart : pupilAvatarParts) {
                    if (pupilAvatarPart.getPartName().equals(pupilAvatarPartStringValues[0])) {
                        result.add(new PupilAvatarPartAndVariant(pupilAvatarPart,
                                Integer.parseInt(pupilAvatarPartStringValues[1])));
                        break;
                    }
                }

            }
        }
        return result;
    }
}

File: sharedModule/src/main/java/ru/nekit/android/qls/pupil/avatar/PupilAvatarConverter.kt
package ru.nekit.android.qls.pupil.avatar

import android.text.TextUtils
import ru.nekit.android.qls.shared.model.Pupil
import ru.nekit.android.qls.shared.model.PupilSex
import java.util.*

class PupilAvatarConverter {

    companion object {

        private val PART_AND_VARIANT_DELIMITER = ":"
        private val DELIMITER = ";"

        fun toString(parts: Array<IPupilAvatarPart>,
                     partVariantPositions: IntArray): String {
            val stringList = ArrayList<String>()
            var index = 0
            for (pupilAvatarPart in parts) {
                stringList.add(String.format("%s%s%s", pupilAvatarPart.name,
                        PART_AND_VARIANT_DELIMITER, partVariantPositions[index]))
                index++
            }
            return TextUtils.join(DELIMITER, stringList)
        }

        fun toPupilAvatarPartAndVariant(pupil: Pupil): List<PupilAvatarPartAndVariant> {
            val result = ArrayList<PupilAvatarPartAndVariant>()
            if (pupil.avatar != null) {
                val pupilAvatarPartStrings = TextUtils.split(pupil.avatar, DELIMITER)
                for (pupilAvatarPartString in pupilAvatarPartStrings) {
                    val pupilAvatarPartStringValues = TextUtils.split(pupilAvatarPartString, PART_AND_VARIANT_DELIMITER)
                    val pupilAvatarParts = (if (pupil.sex == PupilSex.BOY)
                        PupilBoyAvatarPart.values()
                    else
                        PupilGirlAvatarPart.values()) as Array<IPupilAvatarPart>
                    for (pupilAvatarPart in pupilAvatarParts) {
                        if (pupilAvatarPart.name == pupilAvatarPartStringValues[0]) {
                            result.add(PupilAvatarPartAndVariant(pupilAvatarPart,
                                    Integer.parseInt(pupilAvatarPartStringValues[1])))
                            break
                        }
                    }

                }
            }
            return result
        }
    }
}